src/libs/installer/packagemanagercore.cpp

```diff
@@ -648,6 +648,20 @@ int PackageManagerCore::downloadNeededArchives(double partProgressSize)
     ProgressCoordinator::instance()->registerPartProgress(&archivesJob,
         SIGNAL(progressChanged(double)), partProgressSize);
 
+    // Print some progress information on console as well
+    connect(&archivesJob, &DownloadArchivesJob::outputTextChanged, [](const QString &progress) {
+        qDebug().noquote() << progress;
+    });
+    connect(&archivesJob, &DownloadArchivesJob::progressChanged,
+            [lastReported = -10, time = QTime::currentTime()](double progress) mutable {
+        int roughProgress = static_cast<int>(progress * 10) * 10;
+        if (roughProgress > lastReported && time.elapsed() > 3000 || time.elapsed() > 15000) {
```

Comment:
You could add parenthesis to improve readability:
`if ((roughtProgress > lastReported && time.elapsed() > 3000) || time.elapsed() > 15000)` is much faster to read and understand, as you don't have to think about operator precedence etc.

Reply:
Think about && || operator precedence? :) Fixed